xmind/tests/test_workbook_element.py

```diff
@@ -0,0 +1,751 @@
+        _test_el = WorkbookElement()
+
+        _getSheets = self._init_patch_with_name(
+            '_getSheets', 'xmind.core.workbook.WorkbookElement.getSheets',
+            return_value=['sheet1', 'sheet2', 'sheet3'])
+
+        _sheets = _test_el.getSheetByIndex(3)
+
+        _getSheets.assert_called_once_with()
+        self.assertEqual(_sheets, None)
+
+    def test_WorkbookElement_getSheetByIndex_index_greater_then_len_sheets(self):
+        _test_el = WorkbookElement()
+
+        _getSheets = self._init_patch_with_name(
+            '_getSheets', 'xmind.core.workbook.WorkbookElement.getSheets',
+            return_value=['sheet1', 'sheet2', 'sheet3'])
+
+        _sheets = _test_el.getSheetByIndex(4)
+
+        _getSheets.assert_called_once_with()
+        self.assertEqual(_sheets, None)
+
+    def test_WorkbookElement_getSheetByIndex_index_within_len_sheets(self):
+        _test_el = WorkbookElement()
+
+        _getSheets = self._init_patch_with_name(
+            '_getSheets', 'xmind.core.workbook.WorkbookElement.getSheets',
+            return_value=['sheet1', 'sheet2', 'sheet3'])
+
+        _sheets = _test_el.getSheetByIndex(2)
+
+        _getSheets.assert_called_once_with()
+        self.assertEqual(_sheets, 'sheet3')
+
+    def test_WorkbookElement_createSheet_getOwnerWorkbook_is_None(self):
+        _test_el = WorkbookElement()
+        self._remove_patched_function('_getOwnerWorkbook')
+        self._remove_patched_function('_sheet_element_init')
+
+        _getOwnerWorkbook = self._init_patch_with_name(
+            '_getOwnerWorkbook', 'xmind.core.mixin.WorkbookMixinElement.getOwnerWorkbook',
+            return_value=None)
+        _sheet_element_constructor = self._init_patch_with_name(
+            '_sheet_element_constructor', 'xmind.core.workbook.SheetElement',
+            return_value='SheetElement')
+
+        _sheet = _test_el.createSheet()
+
+        _getOwnerWorkbook.assert_called_once_with()
+        _sheet_element_constructor.assert_called_once_with(None, None)
+        self.assertEqual(_sheet, 'SheetElement')
+
+    def test_WorkbookElement_createSheet_is_getOwnerWorkbook(self):
+        _test_el = WorkbookElement()
+        self._remove_patched_function('_getOwnerWorkbook')
+        self._remove_patched_function('_sheet_element_init')
+
+        _getOwnerWorkbook = self._init_patch_with_name(
+            '_getOwnerWorkbook', 'xmind.core.mixin.WorkbookMixinElement.getOwnerWorkbook',
+            return_value='workbook')
+        _sheet_element_constructor = self._init_patch_with_name(
+            '_sheet_element_constructor', 'xmind.core.workbook.SheetElement',
+            return_value='SheetElement')
+
+        _sheet = _test_el.createSheet()
+
+        _getOwnerWorkbook.assert_called_once_with()
+        _sheet_element_constructor.assert_called_once_with(None, 'workbook')
+        self.assertEqual(_sheet, 'SheetElement')
+
+    def test_WorkbookElement_addSheet_index_is_None(self):
```

Comment:
Think about consolidating this group of test into one test
with call 
_test_el.addSheet('sheet') 
_test_el.addSheet('sheet', -1)
_test_el.addSheet('sheet', ) ....